src/udisks2monitor.cpp

```diff
@@ -203,6 +203,42 @@ void UDisks2::Monitor::unmount(const QString &devicePath)
     startMountOperation(devicePath, UDISKS2_FILESYSTEM_UNMOUNT, m_blockDevices->objectPath(devicePath), arguments);
 }
 
+void UDisks2::Monitor::setLabel(const QString &devicePath, const QString &label)
+{
+    if (m_blockDevices->find(devicePath)) {
+        QVariantList args;
+        QVariantMap opts;
+        args << opts;
+
+        const QString dbusMethod = UDISKS2_FILESYSTEM_UNMOUNT;
+
+        QDBusMessage method = QDBusMessage::createMethodCall(
+                            UDISKS2_SERVICE,
+                            m_blockDevices->objectPath(devicePath),
+                            UDISKS2_FILESYSTEM_INTERFACE,
+                            dbusMethod);
+        method.setArguments(args);
+        QDBusPendingCall pendingCall = QDBusConnection::systemBus().asyncCall(method);
+        QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(pendingCall, this);
+        connect(watcher, &QDBusPendingCallWatcher::finished,
+                this, [this, devicePath, label](QDBusPendingCallWatcher *watcher) {
+            if (watcher->isValid() && watcher->isFinished()) {
```

Comment:
Strange pattern in this code, I would think checking watcher for validity and finished state in finished signal handler would be redundant. Anyway, seems to be common to this code, so I guess that's okay.

However, should this handle error state somehow? Like emitting error signal that label could not be set.